pkg/shell/nav.tsx

```diff
@@ -60,60 +117,59 @@ export class CockpitNav extends React.Component {
         };
 
         this.clearSearch = this.clearSearch.bind(this);
+        this.props = props;
     }
 
     componentDidMount() {
-        const self = this;
         const sel = this.props.selector;
         // Click on active menu item (when using arrows to navigate through menu)
         function clickActiveItem() {
             const cur = document.activeElement;
-            if (cur.nodeName === "INPUT") {
-                const el = document.querySelector("#" + sel + " li:first-of-type a");
+            if (cur instanceof HTMLInputElement) {
+                const el = document.querySelector<HTMLElement>("#" + sel + " li:first-of-type a");
                 if (el)
                     el.click();
-            } else {
+            } else if (cur instanceof HTMLElement) {
                 cur.click();
             }
         }
 
         // Move focus to next item in menu (when using arrows to navigate through menu)
         // With arguments it is possible to change direction
-        function focusNextItem(begin, step) {
+        function focusNextItem(begin: number, step: number) {
             const cur = document.activeElement;
-            const all = Array.from(document.querySelectorAll("#" + sel + " li a"));
-            if (cur.nodeName === "INPUT" && all) {
+            const all = Array.from(document.querySelectorAll<HTMLElement>("#" + sel + " li a"));
+            if (cur instanceof HTMLInputElement && all.length > 0) {
                 if (begin < 0)
                     begin = all.length - 1;
                 all[begin].focus();
             } else {
                 let i = all.findIndex(item => item === cur);
                 i += step;
                 if (i < 0 || i >= all.length)
-                    document.querySelector("#" + sel + " .pf-v5-c-text-input-group__text-input").focus();
+                    document.querySelector<HTMLElement>("#" + sel + " .pf-v5-c-text-input-group__text-input")?.focus();
                 else
                     all[i].focus();
             }
         }
 
-        function navigate_apps(ev) {
-            if (ev.keyCode === 13) // Enter
+        const navigate_apps = (ev: KeyboardEvent) => {
+            if (ev.key == "Enter")
```

Comment:
Thanks for making the keys more readable :wink:

Reply:
TS told me that keyCode is deprecated. Nice.